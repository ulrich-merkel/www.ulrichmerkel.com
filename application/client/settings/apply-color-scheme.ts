/**
 * Es6 module for handling theming as component.
 *
 * @file
 * @module
 *
 * @author hello@ulrichmerkel.com (Ulrich Merkel), 2021
 *
 * @see {@link https://blog.risingstack.com/react-js-best-practices-for-2016/}
 */
import { AVAILABLE_COLOR_SCHEMES } from '../../common/state/color-scheme/duck';
import { isBrowser } from '../../common/utils/environment';
import { setDomNodeClassName } from '../utils/dom';

const availableColorSchemes = Object.values(AVAILABLE_COLOR_SCHEMES);

/**
 * Load new theme if necessary and remove old ones.
 *
 * @param {string} colorSchemeSelected - The theme to be applied
 * @returns {void}
 */
export function applyColorScheme(colorSchemeSelected: string): void {
    if (!isBrowser() || !availableColorSchemes.includes(colorSchemeSelected)) {
        return;
    }

    // Check if we should load a new theming css file
    const add = [`color-scheme-${colorSchemeSelected}`];

    // Remove other theming css files
    const remove = availableColorSchemes
        .filter(function filterAvailableThemes(theme) {
            return theme !== colorSchemeSelected;
        })
        .map(function mapAvailableThemes(theme) {
            return `color-scheme-${theme}`;
        });

    setDomNodeClassName('doc-root', add, remove);
}
