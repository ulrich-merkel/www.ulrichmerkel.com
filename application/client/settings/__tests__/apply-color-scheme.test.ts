import { applyColorScheme } from '../apply-color-scheme';

describe('applyColorScheme', function fnDescribe() {
    it('should handle color scheme correctly', function fnIt() {
        expect(applyColorScheme('foo')).toMatchSnapshot();
        expect(applyColorScheme('dark')).toMatchSnapshot();
        expect(applyColorScheme('light')).toMatchSnapshot();
    });
});
