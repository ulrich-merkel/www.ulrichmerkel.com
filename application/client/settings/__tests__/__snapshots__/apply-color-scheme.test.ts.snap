// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applyColorScheme should handle color scheme correctly 1`] = `undefined`;

exports[`applyColorScheme should handle color scheme correctly 2`] = `undefined`;

exports[`applyColorScheme should handle color scheme correctly 3`] = `undefined`;
